src/main/java/platform/codingnomads/co/springdata/example/ddl/joincolumn/User.java

```diff
@@ -17,4 +20,14 @@ public class User {
 
     @Column(name = "name")
     private String name;
+
+    @Column(nullable = false)
+    private String email;
+
+    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
+    @JoinColumn(name = "spouse", unique = true, referencedColumnName = "email", nullable = true, updatable = true)  // updatable=true and nullable=true are default, just making it verbose here for practice.
```

Comment:
Good practice to make everything as explicit as possible. 👍 
Extremely important when you have to work with unknown code (your code will become unknown if you don't touch it for a couple of months).